console-subscriber/src/aggregator.rs

```diff
@@ -615,9 +906,287 @@ impl Task {
+            },
+
+            AttributeUpdateValue::Numeric { val, op, unit } => {
+                let val = *val;
+                let unit = unit.clone();
+                let op = match op {
+                    AttributeUpdateOp::Add => numeric::Op::Add,
+                    AttributeUpdateOp::Sub => numeric::Op::Sub,
+                    AttributeUpdateOp::Ovr => numeric::Op::Ovr,
+                } as i32;
+                PbAttrUpdate {
+                    name: self.name.clone(),
+                    update: Some(PbUpdate::Numeric(PbNumeric { val, op, unit })),
+                }
+            }
+        }
+    }
+}
+
+impl ToProto for AttrValue {
+    type Result = proto::resources::stats::AttrValue;
+
+    fn to_proto(&self) -> Self::Result {
+        use proto::resources::stats::attr_value::Numeric;
+        use proto::resources::stats::attr_value::Val;
+        use proto::resources::stats::AttrValue as PbAttrValue;
+
+        match self {
+            AttrValue::Text(t) => PbAttrValue {
+                val: Some(Val::Text(t.clone())),
+            },
+            AttrValue::Numeric { val, unit } => PbAttrValue {
+                val: Some(Val::Numeric(Numeric {
+                    val: *val,
+                    unit: unit.clone(),
+                })),
+            },
+        }
+    }
+}
+
+impl AttrValue {
+    fn update(&mut self, update: &AttributeUpdateValue) {
+        match (self, update) {
+            (AttrValue::Text(t), AttributeUpdateValue::Text(upd)) => {
+                *t = upd.clone();
+            }
+            (
+                AttrValue::Numeric { ref mut val, .. },
+                AttributeUpdateValue::Numeric { val: upd, op, .. },
+            ) => match op {
+                AttributeUpdateOp::Add => {
+                    let new_val = *val + upd;
+                    *val = new_val;
+                }
+                AttributeUpdateOp::Ovr => {
+                    *val = *upd;
+                }
+                AttributeUpdateOp::Sub => {
+                    let new_val = *val - upd;
+                    *val = new_val;
+                }
+            },
+            _ => panic!("trying to update attributes of different type"),
+        }
+    }
+}
+
+impl From<AttributeUpdateValue> for AttrValue {
+    fn from(upd: AttributeUpdateValue) -> Self {
+        match upd {
+            AttributeUpdateValue::Text(t) => AttrValue::Text(t),
+            AttributeUpdateValue::Numeric { val, unit, .. } => AttrValue::Numeric { val, unit },
+        }
+    }
+}
+
 fn serialize_histogram(histogram: &Histogram<u64>) -> Result<Vec<u8>, V2SerializeError> {
     let mut serializer = V2Serializer::new();
     let mut buf = Vec::new();
     serializer.serialize(histogram, &mut buf)?;
     Ok(buf)
 }
+
+fn total_time(created_at: Option<&SystemTime>, closed_at: Option<&SystemTime>) -> Option<Duration> {
+    closed_at.and_then(|end| created_at.and_then(|start| end.duration_since(*start).ok()))
```

Comment:
nit, take it or leave it: this _might_ be clearer, what do you think?
```suggestion
    let end = closed_at?;
    let start = created_at?;
    end.duration_since(start).ok()
```